### PR TITLE
Lego Developer Guide

### PR DESCRIPTION
**Pull request by Elyash**

### CODE REVIEW

File: developer_guide.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
+# Lego 3 - Developer Guide
+
+This file has 2 main purposes:
+1. Help you develop Lego 3 (core, libs or tests), avoid you from common mistakes and save your time when entering the project.
```
Comment (Steven17D) [inline developer_guide.md:4]: change `avoid you from` to `prevent` common mistakes

File: developer_guide.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+## Notes
+
+### Merging issues
+
+Any contibution of new code to `master` branch should pass the process of PR -> CR.
+this is in purpose to keep the repository clean and order, and make sure the
+contributed code doesn't harm any other code first, and second - do what it
+suppose to do.
+
+Up to now, the permission to merge to `master` can gave by one of us: Ariel Chinn,
```
Comment (Steven17D) [inline developer_guide.md:213]: Use @ annotations instead.

File: developer_guide.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
+# Lego 3 - Developer Guide
```
Comment (Steven17D) [inline developer_guide.md:1]: Run all the file through Grammarly. There are many grammatical and typo.

File: developer_guide.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+### Project
+
+#### Requirements
+
+*Optional*
+
+Lego 3 is a completely new project, written from scratch, but its requirements based on insights from the previous testing infrastructure we used.
+
+Read the requirements for this project [here](https://docs.google.com/document/d/1gSl9_jS_pIAkGtNLJSBKUGRbcOhALKMzv1ewWZSSElg/edit?usp=sharing) under "Requirments".
```
Comment (Steven17D) [inline developer_guide.md:22]: The link here is with editing permissions and this repo is public!!!
Revoke the link, delete the link and preferably delete the link from history.
Fix all the links

File: developer_guide.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+#### Terminology and Structure
+
+*Mandatory*
+ 
+Whether you already familiar with Lego or not, it is important to understand the new terminology of Lego 3 and its structure.
+
+Read Lego 3 terminology [here](https://docs.google.com/document/d/1gSl9_jS_pIAkGtNLJSBKUGRbcOhALKMzv1ewWZSSElg/edit?usp=sharing) under "Terminology".
```
Comment (Steven17D) [inline developer_guide.md:30]: Fix link

File: developer_guide.md (hunk 4 above)
Comment (Elyash) [reply developer_guide.md:22]: Check again

Comment (Elyash) [reply developer_guide.md:22]: Check again

File: developer_guide.md (hunk 5 above)
Comment (Elyash) [reply developer_guide.md:30]: Same as above

File: developer_guide.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+#### RPyC
+
+*Mandatory*
+
+RPyC is a Python library for Python connection to remote devices.
+
+It is one of the basic stones of Lego 3.
+
+Read [here](https://rpyc.readthedocs.io/en/latest/index.html) the "Tutorial" and "Documentation" paragraphes.
```
Comment (Steven17D) [inline developer_guide.md:70]: typo -> paragraphs

**Pull request merged**